### ISSUE TITLE
Flash of unstyled content when using next export

### ISSUE THREAD

**Issue by sinafath** [mantinedev/mantine#4499]
### What package has an issue

@mantine/core

### Describe the bug

address website:[link](https://friendly-bardeen-8hfq-p8au.iran.liara.run/)
i used [next-pages-template](https://github.com/mantinedev/next-pages-template)

document:
```
import { createGetInitialProps } from '@mantine/next'
import Document, { Head, Html, Main, NextScript } from 'next/document'

const getInitialProps = createGetInitialProps()

export default class _Document extends Document {
  static getInitialProps = getInitialProps

  render() {
    return (
      <Html dir="rtl">
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}
```
app:
```

import  { AppProps } from "next/app";
import Head from "next/head";
import {
  MantineProvider,
  ColorScheme,
  createEmotionCache,
} from "@mantine/core";
import { Notifications } from "@mantine/notifications";
import rtlPlugin from "stylis-plugin-rtl";


const rtlCache = createEmotionCache({
  key: "mantine-rtl",
  stylisPlugins: [rtlPlugin],
});
export default function App(props: AppProps & { colorScheme: ColorScheme }) {
  const { Component, pageProps } = props;


  return (
    <>
      <Head>
        <title>Mantine next example</title>
        <meta
          name="viewport"
          content="minimum-scale=1, initial-scale=1, width=device-width"
        />
      </Head>

        <MantineProvider
          withGlobalStyles
          withNormalizeCSS
          emotionCache={rtlCache}
                theme={{
            dir: "rtl",    }}
        >
          <Component {...pageProps} />
          <Notifications />
        </MantineProvider>
    </>
  );
}

```
evething works on local but when I uae next export i hvae a Flash of unstyled content

### What version of @mantine/hooks page do you have in package.json?

latest

### If possible, please include a link to a codesandbox with the reproduced problem

https://friendly-bardeen-8hfq-p8au.iran.liara.run

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by chewhx** [comment]
Try updating your `_document.tsx` based on the example code provided: https://mantine.dev/theming/emotion-cache/#server-side-rendering-with-custom-cache

**Comment by aliasadolahi74** [comment]
> Try updating your `_document.tsx` based on the example code provided: https://mantine.dev/theming/emotion-cache/#server-side-rendering-with-custom-cache

It's not working on next.js 13.4 because there is no document and app file in the app directory anymore

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake and it is still an issue in version 7.0, please reopen it.
